core/src/autogluon/core/scheduler/reporter.py

```diff
@@ -97,6 +98,53 @@ def __repr__(self):
         return reprstr
 
 
+class MODistStatusReporter(DistStatusReporter):
+    """Report status through the training of a multi-objective scheduler.
+
+    Example
+    -------
+    >>> @autogluon_method
+    >>> def train_func(config, reporter):
+    ...     reporter(accuracy=1, f_score=1, training_iters=4)
+    """
+    def __init__(self, objectives, weights, scalarization_opts, remote=None):
+        super().__init__(remote)
+        self.objectives = objectives
+        self.weights = weights
+        self.scalarization_options = scalarization_opts
+    
+    def __call__(self, **kwargs):
+        """Report updated training status.
+        Pass in `done=True` when the training job is completed.
+
+        Args:
+            kwargs: Latest training result status. Reporter requires access to
+            all objectives of interest.
+
+        Example
+        _______
+        >>> reporter(accuracy=1, f_score=1, training_iters=4)
+        """
+        try:
+            v = np.array([kwargs[k] for k in self.objectives])
+        except KeyError:
+            raise KeyError("Reporter requires accesss to all objective values.\
+                Please ensure you return all required values.")
+
+        if self.scalarization_options["algorithm"] == "random_weights":
```

Comment:
Is this a good name? The weights are fixed here. I suppose the difference between this and Parego is rather how the weights are combined with the scalars?